core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt

```diff
@@ -63,7 +63,9 @@ object UniqueTriggerActivation {
                 if (limit!=null && limit <= civInfo.units.getCivUnits().count { it.name==unitName })
                     return false
 
-                val placedUnit = civInfo.units.addUnit(unitName, chosenCity) ?: return false
+                val placedUnit = if (city != null || tile == null)
```

Comment:
Sorry what? If city is null and tile is null then add next to chosen city? Something here is off